docs: document how to use `sudo` if modifying an object (#2928)
python-gitlab/python-gitlab: docs/api-usage.rst

   p = gl.projects.create({'name': 'awesome_project'}, sudo='user1')

.. warning:: 

>    When using ``sudo``, its usage is not remembered. What this means is that if
>    you use ``sudo`` to retrieve an object and then later use ``save()`` to
>    modify the object it will not use ``sudo``. In that case you should do
>    ``save(sudo='user1')``

Reviewer: Could we maybe shorten the sentence a bit as it's inside an admonition block, so we can also mention why this is needed?

```suggestion
   When using ``sudo``, its usage is not remembered. If you use ``sudo`` to retrieve an object
   and then later use ``save()`` to modify the object, it will not use ``sudo``.  You should use
   ``save(sudo='user1')`` if you want to perform subsequent actions as the  user.
```

Author: Done.

Updating with ``sudo``
----------------------

An example of how to ``get`` an object (using ``sudo``), modify the object, and

> then ``save`` the object (using ``sudo``)

Reviewer: ```suggestion
then ``save`` the object (using ``sudo``):
```

Author: Done

.. code-block:: python

   group = gl.groups.get('example-group')
   notification_setting = group.notificationsettings.get(sudo='user1')
   notification_setting.level = gitlab.const.NOTIFICATION_LEVEL_GLOBAL
   # Must use `sudo` again when doing the save.

>    notifcation_setting.save(sudo='user1')

Reviewer: ```suggestion
   notification_setting.save(sudo='user1')
```
